client/src/containers/SigninForm.jsx

```diff
@@ -28,14 +31,30 @@ const SigninForm = () => {
             />
             <PasswordInput
               value={password}
-              onChange={event => setPassword(event.target.value)}
+              onChange={(event) => setPassword(event.target.value)}
             />
-            <SubmitButton>Sign in</SubmitButton>
+            {props.error ? (<p> { props.error } </p>) : null}
+            <SubmitButton disabled={props.loading}>Sign in</SubmitButton>
```

Comment:
Here you can also pass loading as prop to the submit button and in the submit button component you can display a loader within the button, so that user can get a feel okay something is happening, its not stuck. Its just a ui suggestion.

Reply:
makes sense, I will do that.